deps/rabbitmq_management/src/rabbit_mgmt_wm_queue_get.erl

```diff
@@ -47,34 +48,124 @@ accept_content(ReqData, Context) ->
+                                    end,
+                            CTag = <<"ctag">>,
+                            Reply = start_subscription_gets(
+                              Count, Ch, Q, CTag, Offset, Enc, Trunc),
+                            rabbit_mgmt_util:reply(remove_delivery_tag(Reply),
+                               ReqData, Context)
+                        end)
+              end);
+        _ ->
+            rabbit_mgmt_util:with_decode(
+              [ackmode, count, encoding], ReqData0, Context,
+              fun([AckModeBin, CountBin, EncBin], Body, ReqData) ->
+                    rabbit_mgmt_util:with_channel(
+                        VHost, ReqData, Context,
+                        fun (Ch) ->
+                            AckMode = list_to_atom(binary_to_list(AckModeBin)),
+                            Count = rabbit_mgmt_util:parse_int(CountBin),
+                            Enc = case EncBin of
+                                      <<"auto">>   -> auto;
+                                      <<"base64">> -> base64;
+                                      _            -> throw({error, <<"Unsupported encoding. Please use auto or base64.">>})
+                                  end,
+                            Trunc = case maps:get(truncate, Body, undefined) of
+                                        undefined -> none;
+                                        TruncBin  -> rabbit_mgmt_util:parse_int(
+                                                       TruncBin)
+                                    end,
+                            Reply = basic_gets(Count, Ch, Q, AckMode, Enc,
+                                               Trunc),
+                            maybe_return(Reply, Ch, AckMode),
+                            rabbit_mgmt_util:reply(remove_delivery_tag(Reply),
+                               ReqData, Context)
+                        end)
+              end)
+    end.
+
+start_subscription_gets(Count, Ch, Queue, CTag, Offset, Enc, Trunc) ->
+    qos(Ch, Count),
+    subscribe(Ch, Queue, false, Offset, CTag),
+    Replies = subscription_gets(Count, Ch, Queue, CTag, Offset, Enc, Trunc),
+    cancel_subscription(Ch, CTag),
+    Replies.
+
+subscription_gets(0, _Ch, _Queue, _CTag, _Offset, _Enc, _Trunc) ->
+    [];
+subscription_gets(Count, Ch, Queue, CTag, Offset, Enc, Trunc) ->
+    case subscription_get(Ch, Enc, Trunc) of
+        none -> [];
+        Reply -> [Reply | subscription_gets(Count - 1, Ch, Queue, CTag, Offset, Enc, Trunc)]
+    end.
 
+subscription_get(Ch, Enc, Trunc) ->
+    receive 
+        {#'basic.deliver'{redelivered = Redelivered,
+                         exchange = Exchange,
+                         routing_key = RoutingKey,
+                         delivery_tag = DeliveryTag,
+                         consumer_tag = ConsumerTag},
+        #amqp_msg{props = Props, payload = Payload}} ->
+            ok = amqp_channel:cast(Ch, #'basic.ack'{delivery_tag = DeliveryTag,
```

Comment:
I dont think we want to ack the message. If the stream has 20 messages, and we ask for 10 (prefetch = 10) when we ack the first message, the broker will send the 11th message. (Prefetch count means the max number of outstanding ie unacknowledged messages)

Reply:
This feature should not arguably use AMQP 0-9-1 at all. And the original feature should never have used an AMQP 0-9-1 Erlang client to begin with.

Reply:
Yes it would probably be better to use `rabbit_queue_type` directly to get a message off the stream.

Reply:
I dont say that an amqp connection should be used to implement this feature, just want to mention the aspect, or implicit benefit that the server side of an amqp connection does additional authorization checks.

Would like to cite the recent attempt for the delete queue endpoint to use an internal API https://github.com/rabbitmq/rabbitmq-server/pull/9550 which was reverted https://github.com/rabbitmq/rabbitmq-server/pull/10062#issuecomment-1864117758. Maybe can you recall what was the difficulty, why the path outlined in #10062 was abandoned?

Or maybe that restriction is not applicable for a simple reading from a queue?

Reply:
Sounds like a way to not move anywhere.. why would not using the 0-9-1 protocol be ok expect it's maybe not 100% optimal? This solves a real world problem with minimal change.

Reply:
Dear Core Team! Do you know the difficulties of or things to consider when converting the Management plugin to not use an AMQP client, eg in context of #9550 and #10062? This would be helpful to move forward with this PR either using an amqp client or not.